File: frontend/src/api/ai.ts
import axios from "axios";

const API_URL = import.meta.env.VITE_API_URL;

export interface StressPredictionDay {
  day: string;
  stressLevel: number;
}

export interface StressPredictionResponse {
  daily_stress: StressPredictionDay[];
  weekly_stress_score: number;
  risk_level: "low" | "medium" | "high";
  peak_stress_day: string;
  explanation: string;
}

export const getStressPrediction = async (
  token: string,
  weeklyLoad: { day: string; hours: number; deadlines: number }[]
) => {
  const response = await axios.post<StressPredictionResponse>(
    `${API_URL}/ai/stress-prediction`,
    { weekly_load: weeklyLoad },
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
};

File: loadalert_backend/app/schemas/ai.py
from pydantic import BaseModel
from datetime import date

# Stress Prediction
class StressPredictionDayInput(BaseModel):
    day: str
    hours: int
    deadlines: int

class StressPredictionRequest(BaseModel):
    weekly_load: list[StressPredictionDayInput]

class StressPredictionDayOutput(BaseModel):
    day: str
    stressLevel: int  # 0–100

class StressPredictionResponse(BaseModel):
    daily_stress: list[StressPredictionDayOutput]
    weekly_stress_score: int  # 0–100
    risk_level: str           # "low" | "medium" | "high"
    peak_stress_day: str
    explanation: str

# Priorities
class PriorityTaskInput(BaseModel):
    id: int
    title: str
    due_date: date
    estimated_effort: int
    importance_level: str

class PrioritiesRequest(BaseModel):
    tasks: list[PriorityTaskInput]

class PriorityTaskOutput(BaseModel):
    id: int
    title: str
    rank: int
    reason: str
    estimated_effort: int
    due_date: date

class PrioritiesResponse(BaseModel):
    priorities: list[PriorityTaskOutput]

# Stress Contributors
class StressContributorInput(BaseModel):
    id: int
    title: str
    due_date: date
    estimated_effort: int
    importance_level: str

class StressContributorsRequest(BaseModel):
    deadlines: list[StressContributorInput]

class StressContributorOutput(BaseModel):
    id: int
    title: str
    contribution: int  # percentage 0-100
    due_date: date

class StressContributorsResponse(BaseModel):
    contributors: list[StressContributorOutput]
    max_contribution: int
